package requires newer Go version go1.24 (application built with go1.23)

### Problem statement

Running `mockery` with sources from module that define go 1.24 as minimal Go version yields:

```
2025-04-10T10:16:35.097629000+02:00 INF Starting mockery config-file=/Users/X/path/.mockery.yaml version=v3.0.0
2025-04-10T10:16:35.098266000+02:00 INF Parsing configured packages... version=v3.0.0
2025-04-10T10:16:37.898622000+02:00 ERR encountered error when loading package error="/Users/X/path/sdk/controlplane.go:1:1: package requires newer Go version go1.24 (application built with go1.23)" version=v3.0.0
2025-04-10T10:16:37.898749000+02:00 ERR encountered error when loading package error="-: This application uses version go1.23 of the source-processing packages but runs version go1.24 of 'go list'. It may fail to process source files that rely on newer language features. If so, rebuild the application using a newer version of Go." version=v3.0.0
2025-04-10T10:16:37.898923000+02:00 ERR unable to parse packages error="error occurred when loading packages" version=v3.0.0
2025-04-10T10:16:37.899083000+02:00 FTL app failed error="error occurred when loading packages" version=v3.0.0
```

### Proposed solution

Bump minimum Go version in `go.mod` to 1.24.

--- a/.github/workflows/tag-and-release.yml
+++ b/.github/workflows/tag-and-release.yml
@@ -72,7 +72,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: '1.21'
+          go-version: 'stable'
 
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2

